packages/template-retail-react-app/app/pages/cart/index.jsx

```diff
@@ -86,6 +86,77 @@ const Cart = () => {
 
     const {data: customer} = useCurrentCustomer()
     const {customerId, isRegistered} = customer
+
+    /***************** Product Bundles ************************/
+
+    // We use the `products` object to reference products by itemId instead of productId
+    // Since with product bundles, even though the parent productId is the same,
+    // variant selection of the bundle children can be different,
+    // and require unique references to each product bundle
+    const [productsByItemId, setProductsByItemId] = useState({})
+
+    const bundleChildVariantIds = []
+    basket?.productItems?.forEach((productItem) => {
+        productItem?.bundledProductItems?.forEach((childProduct) => {
+            bundleChildVariantIds.push(childProduct.productId)
+        })
+    })
+
+    const {data: bundleChildProductData} = useProducts(
+        {
+            parameters: {
+                ids: bundleChildVariantIds,
+                allImages: false,
+                expand: ['availability', 'variations'],
+                select: '(data.(id,inventory))'
+            }
+        },
+        {
+            enabled: bundleChildVariantIds?.length > 0,
+            keepPreviousData: true,
+            select: (result) => {
+                return result?.data?.reduce((result, item) => {
+                    const key = item.id
+                    result[key] = item
+                    return result
+                }, {})
+            }
+        }
+    )
+
+    // Seting up productsByItemId state where key is itemId and value is the product data
+    const updateProductsByItemId = {}
+    basket?.productItems?.forEach((productItem) => {
```

Comment:
Because this logic performs the expensive call `JSON.stringify`, I think do it only if `basket.productItems` changes (and also wait for `products` and `bundleChildProductData` to exist).